### PR TITLE
[sc-28688] Add template for AWS S3

### PR DESCRIPTION
**Pull request by r-rakshith**
This PR adds a template for AWS S3 as an SoR.

Shortcut story

[sc-28688](https://app.shortcut.com/sgnl/story/28688)

### CODE REVIEW

**Comment by marc-sgnl** [comment]
@r-rakshith I know it's a little strange, but do want to include any entities/relationships in this catalog template? For sure the icon/auth/config makes sense, but there's probably only a slim chance that the entities do as well. Would love your thoughts.

File: aws-s3.sgnl.yaml
Comment (richardxgao) [inline aws-s3.sgnl.yaml:91]: Do User -> Group related relationships need to be created?

File: aws-s3.sgnl.yaml
Comment (richardxgao) [inline aws-s3.sgnl.yaml:26]:
```suggestion
```

**Comment by r-rakshith** [comment]
> @r-rakshith I know it's a little strange, but do want to include any entities/relationships in this catalog template? For sure the icon/auth/config makes sense, but there's probably only a slim chance that the entities do as well. Would love your thoughts.

I'm willing to skip on the entities and relationships in the template. It's included right now as an example to provide guidance on a couple of things like
- externalId of an entity must match the corresponing `.csv` file in the S3 bucket
- the file containing membership data must have a column that represents a uniqueId attribute

As long as these are captured in the help docs, I start removing entities and relationships.

File: aws-s3.sgnl.yaml
Comment (r-rakshith) [reply aws-s3.sgnl.yaml]: Removing the example entities and relationships.

**Comment by marc-sgnl** [comment]
> I'm willing to skip on the entities and relationships in the template. It's included right now as an example to provide guidance on a couple of things like

I'm torn on it, maybe we just do one entity (like User) that illustrates your point? I know it has been challenging in the past to get people to read our docs, so I do like having some cues that a smart user can leverage to be successful quickly.

**Comment by r-rakshith** [comment]
> > I'm willing to skip on the entities and relationships in the template. It's included right now as an example to provide guidance on a couple of things like
> 
> I'm torn on it, maybe we just do one entity (like User) that illustrates your point? I know it has been challenging in the past to get people to read our docs, so I do like having some cues that a smart user can leverage to be successful quickly.

Sounds like a plan! 
I'll keep the `User` entity and remove the other two.

**Review by richardxgao — APPROVED** [review]
LGTM pending @marc-sgnl approval.

**Comment by marc-sgnl** [comment]
Approved from my side, thanks so much @r-rakshith and thanks for the support reviewing @richardxgao

**Review by marc-sgnl — APPROVED** [review]
LGTM

**Pull request merged**
